Add linking to Events cards

tomayyeung opened the pull request:
With this PR:

- Event cards are all given `id`s e.g. `2025-02-21-staging` that can be linked to
- On `/events` page load, if there is a URL fragment pointing to an event `id`, that card is centered
- For all Markdown files, when a `slug` is not specified, the filepath is injected as a fallback (matching the behavior as described on the [web platform docs](https://compsigh.club/docs/web-platform/how-to-publish#slug))

Other changes include:

- Modularization of Markdown compiling
- Beginning work on better documentation and a rewrite of logic around Markdown files
  - See the [undesired behavior around event slugs noted here](https://github.com/compsigh/web/pull/26#issuecomment-2764410095)

edwardshturman commented:
aghh i don't like the implementation strat yet

- `<EventList />` calls `scrollIntoView()` for the latest event, overriding the initial page load scroll position
- `idStartDate` in some cases is wrong bcs [`toISOString()` is always UTC](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString). e.g. the `id` for the BLOOM event card: `2025-03-01-BLOOM`
- more of a nit but `event.title` is still kinda yucky with URI encoding. would prefer to use the filename (which is already optimized to be a slug) if we can somehow get that from one of the Markdown helper functions i wrote

open to ideas

tomayyeung commented:
- fixed; if there's a hash in the URL, `<EventList />` doesn't call `scrollIntoView()`
- fixed; scuffed, uses `sv-SE` (sweden) locale to get the desired yyyy-mm-dd format
  - only alternative i can think of that doesn't look so random is manually piecing the string together
- fixed; scuffed, it takes the last file in the folder for the event, this typically grabs the  correct one, but rn i can't think of a good way to get the desired file, since `EventDetails` only stores `start` date, not its associated Markdown file
  - `event.event_details.link` doesn't work, it's just a full URL
  - either store its associated md file, or change the file structure? that's what i can think of rn

edwardshturman commented:
First is fine

Second is...fine lol

Third def does not work. Days can contain multiple events (and an event can span over multiple days). The strat will probably be inspired by something in [`/[...slug]/page.tsx`](https://github.com/compsigh/web/blob/main/app/%5B...slug%5D/page.tsx). Storing the filename (to avoid rereads)

edwardshturman commented:
from dms:

> idea: we already have a `slug` field, but rn it represents overrides to the filesystem-based route. what if we just populate that value *with the filepath* at build time if none was specified in frontmatter. then extract substring after the last `/`

edwardshturman commented:
from dms:

> what puzzles the shit out of me rn: giving `app/events/2025-02-21/staging.md` a `slug` of `amogus` maps *both* `events/2025-02-21/staging` *and* `amogus` as valid slugs
> BUT
> `app/docs/support.md` `slug = "support"` → `support` works, but `docs/support` doesn't (intended behavior)

edwardshturman commented:
Noticing an issue with our `scrollIntoView()` bypass: we set the `id` on just the `<p>` tag representing the date. When you navigate to the link, instead of scrolling *the event card* into center view, it scrolls to the date aligned to the top of the page. Working on this now

<img width="2024" alt="image" src="https://github.com/user-attachments/assets/2abd6824-bba1-4b5e-8ef3-0bbcd05f31d7" />

edwardshturman commented:
We didn't wanna introduce more markup soup lol (there's like 3 wrapper divs 😭) but it's fixed; much better

edwardshturman commented:
> from dms:
> 
> > what puzzles the shit out of me rn: giving `app/events/2025-02-21/staging.md` a `slug` of `amogus` maps _both_ `events/2025-02-21/staging` _and_ `amogus` as valid slugs
> > BUT
> > `app/docs/support.md` `slug = "support"` → `support` works, but `docs/support` doesn't (intended behavior)

I wanna tackle this but probably in a separate PR. I think we achieved what we were going for here 🫡

edwardshturman commented:
Updated PR desc
